src/compiler/sys.ts

```diff
@@ -462,6 +462,31 @@ namespace ts {
                 return useCaseSensitiveFileNames ? path : path.toLowerCase();
             }
 
+            const enum FileSystemEntryKind {
+                File,
+                Directory
+            }
+            function fileSystemEntryExists(path: string, entryKind: FileSystemEntryKind): boolean {
+                try {
+                    const stat = _fs.statSync(path);
+                    switch (entryKind) {
+                        case FileSystemEntryKind.File: return stat.isFile();
+                        case FileSystemEntryKind.Directory: return stat.isDirectory();
+                    }

```

Comment:
moving it out of try wont make any difference since `try` prevents optimization for the entire function. I've considered extracting this into a separate function but given that I don't see any perf differences when compiling Monaco I don't think we need to do this.